src/databricks/labs/ucx/hive_metastore/mapping.py

```diff
@@ -103,14 +138,70 @@ def skip_table(self, backend: StatementExecutionBackend, schema: str, table: str
         except BadRequest as br:
             logger.error(br)
 
-    def skip_schema(self, backend: StatementExecutionBackend, schema: str):
+    def skip_schema(self, schema: str):
         # Marks a schema to be skipped in the migration process by applying a table property
         try:
-            backend.execute(f"ALTER SCHEMA `{schema}` SET DBPROPERTIES('{self.UCX_SKIP_PROPERTY}' = true)")
+            self._backend.execute(f"ALTER SCHEMA `{schema}` SET DBPROPERTIES('{self.UCX_SKIP_PROPERTY}' = true)")
         except NotFound as nf:
             if "[SCHEMA_NOT_FOUND]" in str(nf):
                 logger.error(f"Failed to apply skip marker for Schema {schema}. Schema not found.")
             else:
                 logger.error(nf)
         except BadRequest as br:
             logger.error(br)
+
+    def _validate_source_table(self, table: Table) -> Table | None:
+        result = self._backend.fetch(f"SHOW TBLPROPERTIES `{table.database}`.`{table.name}`")
+        for value in result:
+            if value["key"] == "upgraded_to":
+                logger.info(f"{table.key} is set as upgraded to {value['value']}")
```

Comment:
Bug: if the referred table in upgraded_to doesn't exist or is incorrectly migrated, it's get skipped without a notice.

Call ws.tables.get() and compare the "upgraded_from" property.

Reply:
Can I rely on the snapshot?

Reply:
That's a good question. I'm not sure how to handle such cases. In any case we have to remove the upgraded_to as it will block sync operation.

Reply:
Let's consider an exception log/table that allows you to fix and rerun. I'm also considering "dry run" option.
That is another PR.

Reply:
It is easy to find out if from_table is pointing nowhere, but Ill be reluctant to touch it. The user now can use the revert command to revert this parameter.